packages/expo-notifications/android/src/main/java/expo/modules/notifications/push/TokenDispatcher/engines/BareEngine.java

```diff
@@ -0,0 +1,15 @@
+package expo.modules.notifications.push.TokenDispatcher.engines;
+
+import android.content.Context;
+
+public class BareEngine implements Engine {
```

Comment:
Let's refactor the engines logic so it doesn't include a class that basically does nothing. Doing nothing should be a default way of doing things. Someone should be able to optionally inject some logic into token handling code (some way of registering listeners), but in general `BareEngine` is what shouldn't be an extra class, but the default implementation.

Reply:
I can rename it to DefaultEngine if you want. I think it's better than the next conditional statement in code.

Reply:
If instead of engines which work "one or the other" we would organize token processing as in publisher/subscriber we could have multiple subscribers to one token and `TokenDispatcher` would not have to depend on the exact classes. This would also make the future split of `expo-notifications` easier, if we would have `TokenSubscriber` interface extracted to `unimodules-notifications-interface`